packages/shared/lib/services/connection.service.ts

```diff
@@ -618,21 +602,59 @@ class ConnectionService {
                 connection,
                 providerConfig: config,
                 template: template as ProviderTemplateOAuth2,
-                activityLogId,
-                environment_id: environmentId,
-                instantRefresh,
-                logAction: action,
-                logContextGetter
+                environment_id: environment.id,
+                instantRefresh
             });
 
-            if (!success) {
-                return { success, error, response: null };
+            if (!success && error) {
+                const log: ActivityLog = {
+                    level: 'error' as LogLevel,
+                    success: false,
+                    action: LogActionEnum.AUTH,
+                    start: Date.now(),
+                    end: Date.now(),
+                    timestamp: Date.now(),
+                    connection_id: connectionId,
+                    provider_config_key: providerConfigKey,
+                    provider: config.provider,
+                    session_id: '',
+                    environment_id: environment.id,
+                    operation_name: 'Auth'
+                };
+
+                const logMessage: ActivityLogMessage = {
+                    environment_id: environment.id,
+                    level: 'error',
+                    content: error?.message || 'Failed to refresh credentials',
+                    timestamp: Date.now()
+                };
+
+                const activityLogId = await createActivityLogAndLogMessage(log, logMessage);
+
+                const logCtx = await logContextGetter.create(
+                    { id: String(activityLogId), operation: { type: 'token' }, message: 'Token refresh error' },
+                    {
+                        account,
+                        environment,
+                        integration: config ? { id: config.id, name: config.unique_key, provider: config.provider } : undefined,
+                        connection: { id: connection.id, name: connection.connection_id }
+                    }
+                );
+
+                await logCtx.error('Failed to refresh credentials', error);
+                await logCtx.failed();
```

Comment:
I feel like we should log the success here because `getConnectionCredentials` is called in 5 differents places and only the getConnectionWeb is logging the success

Reply:
okay so if I understand well we don't log all refresh, only the one that were forced, right? 👍🏻

Reply:
That is correct